improve default configuration (#1508)
bareos/bareos: docs/manuals/source/Configuration/Director.rst

.. config:option:: dir/fileset/include/options/HardLinks

   :type: yes|no
   :default: no

>    When enabled, this directive will cause hard links to be backed up. 

Reviewer: ```suggestion
   When enabled, this directive will cause hard links to be backed up as hard links.
```

Author: Thanks a lot for the suggestion, will rework the whole paragraph.

   However, the File daemon keeps track of hard linked files and
   will backup the data only once. When you have a file with multiple hardlinks 
   and the first hardlink encountered gets excluded, then no actual file data is 
   send but the backup is still finishes without errors, which normally would create
   an error on restore. In such a case the hardlink information will be dropped and 

>    not restored.

Reviewer: For each set of hardlinks, the file daemon will only backup the file contents once -- when it encountered the first file of that set -- and only backup meta data and a reference to the first file for each other file in that set.  This means that you should only ever restore _all_ files in that set of hardlinks at once or not restore any of them.
If you were to restore a file inside that set, which was not the file with the contents attached, then bareos will not restore its data, but instead just try to link with the file it references and update its meta data.
This means that the newly restored file might not actually have the same contents as when it was created.

Author: Thanks a lot for the suggestion, will rework the whole paragraph.

>    Also be aware that the process of keeping track of the hard links can be quite 

Reviewer: ```suggestion
   Be aware that the process of keeping track of the hard links can be quite 
```

.. config:option:: dir/fileset/include/options/HardLinks

   :type: yes|no
   :default: no 

   New default since :sinceVersion:`23.0.0: Bareos > 23.0.0`

>    When disabled Bareos will most likely backup more data and the system will 
>    not be restored identically to the original. The fact that file is a hardlink
>    will be missing.

Reviewer: ```suggestion
   When disabled, Bareos will backup each file separately from other hardlinked files
   and restore them as unrelated files as well.  The fact that the files were
   hardlinks will be missing.
```

   When enabled, this directive will cause hard links to be backed up as hard links.
   For each set of hardlinks, the file daemon will only backup the file contents once --
   when it encounters the first file of that set -- and only backup meta data and a
   reference to that first file for each subsequent file in that set. 

   Be aware that the process of keeping track of the hard links can be quite 
   expensive if you have lots of them (tens of thousands or more). Backups become 
   very long and the File daemon will consume a lot of CPU power checking hard links.
   In such a case, set :config:option:`dir/fileset/include/options/HardLinks = no`

>    and hard links will not be backed up.

Reviewer: Do you have any expierences with this ?  I tried creating 600k sets of 3 hardlinked files each and the version that with `HardLinks=yes` was almost twice as fast as the version with `Hardlinks=no`.  Im not sure if this is still true (if it ever was!)

.. config:option:: dir/fileset/include/options/HardLinks

   :type: yes|no

>    :default: no 

Reviewer: ```suggestion
   :default: no
```

>    New default since :sinceVersion:`23.0.0: Bareos >= 23.0.0`

Reviewer 2: ```suggestion
   Since :sinceVersion:`23.0.0: fileset keep hard links option switched from yes to no`
   the default is :strong:`no`, before it has been :strong:`yes`.
```
sinceVersion writes "Bareos >= xxx" automatically. The text after the column will be written to the index.

This information is now twice in this section. It might be enough, to combine this with the note at the end.

Is this information also provided as breaking (?) change in the changelog?

   When disabled, Bareos will backup each file individually and restore them as

>    unrelated files as well. The fact that the files were hard links will be 

Reviewer: ```suggestion
   unrelated files as well. The fact that the files were hard links will be
```

   lost.

   When enabled, this directive will cause hard links to be backed up as hard
   links. For each set of hard links, the file daemon will only backup the file
   contents once -- when it encounters the first file of that set -- and only
   backup meta data and a reference to that first file for each subsequent file

>    in that set. 

Reviewer: ```suggestion
   in that set.
```

> 
>    Be aware that the process of keeping track of the hard links can be quite 
>    expensive if you have lots of them (tens of thousands or more). Backups 
>    become very long and the File daemon will consume a lot of CPU power 
>    checking hard links.

Reviewer: ```suggestion

   Be aware that the process of keeping track of the hard links can be quite
   expensive if you have lots of them (tens of thousands or more). Backups
   become very long and the File daemon will consume a lot of CPU power
   checking hard links.
```

   See related performance option like :config:option:`dir/director/OptimizeForSize`

   .. note::

      If you created backups with :config:option:`dir/fileset/include/options/HardLinks = yes`
      you should only ever restore all files in that set of hardlinks at once
      or not restore any of them.
      If you were to restore a file inside that set, which was not the file
      with the contents attached, then Bareos will not restore its data, but
      instead just try to link with the file it references and restore its meta
      data.
      This means that the newly restored file might not actually have the same
      contents as when it was backed up.

>  
>    .. note::
>    

Reviewer: ```suggestion

   .. note::

```

>       Before version 23.0.0 the default was :strong:`yes`:

Reviewer: ```suggestion
      Before version 23.0.0 the default was :strong:`yes`.
```

.. config:option:: dir/fileset/include/options/HardLinks

   :type: yes|no
   :default: no

   .. note::

>       Since :sinceVersion:`23.0.0: fileset keep hard links option switched from
>       yes to :strong:`no`.

Reviewer: The text between `:` and the closing ` is not printed. It is only used for searching I think.
```suggestion
      In previous versions the default was :strong:`yes`, but since :sinceVersion:`23.0.0: HardLinks = no` the default is :strong:`no`. 
```
